src/D2L.CodeStyle.Analyzers/Language/ParamNamesShouldMatchInterfaceAnalyzer.cs

```diff
@@ -0,0 +1,87 @@
+﻿using System;
+using System.Collections.Immutable;
+using System.Linq;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Diagnostics;
+
+namespace D2L.CodeStyle.Analyzers.Language {
+	[DiagnosticAnalyzer( LanguageNames.CSharp )]
+	internal sealed class ParamNamesShouldMatchInterfaceAnalyzer : DiagnosticAnalyzer {
+		public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray.Create(
+			Diagnostics.InterfaceImplementationParamNameMismatch
+		);
+
+		public override void Initialize( AnalysisContext context ) {
+			context.EnableConcurrentExecution();
+			context.RegisterCompilationStartAction( RegisterAnalyzer );
+		}
+
+		private static void RegisterAnalyzer( CompilationStartAnalysisContext context ) {
+			context.RegisterSyntaxNodeAction(
+				ctx => AnalyzeClassDefinition(
+					ctx,
+					( INamedTypeSymbol )ctx.SemanticModel.GetDeclaredSymbol( ctx.Node )
+				),
+				SyntaxKind.ClassDeclaration
+			);
+		}
+
+		private static void AnalyzeClassDefinition(
+			SyntaxNodeAnalysisContext context,
+			INamedTypeSymbol typeSymbol
+		) {
+			if( !typeSymbol.Interfaces.Any() ) {
+				return;
+			}
+
+			ImmutableArray<IMethodSymbol> interfaceMethods = CollectInterfaceMethods( typeSymbol );
+			foreach( var interfaceMethod in interfaceMethods ) {
+				IMethodSymbol implMethod = (IMethodSymbol)typeSymbol.FindImplementationForInterfaceMember( interfaceMethod );
+
+				// In the middle of coding and haven't implemented an interface method yet

```

Comment:
Its possible to add something to the spec for this

Reply:
Tru, that's where I saw it